### ISSUE TITLE
Evaluate current compile caching reuse and speedup

### ISSUE THREAD

**Issue by sducouedic** [torch-spyre/spyre-inference#302]
- Evaluate the current speedup obtained from repeated runs
- Check and understand the structure of compiled artifacts, and which elements are reused

**Comment by sducouedic** [comment]
I ran the following python script (`test_matmul_caching.py`) twice, and verified the produced compiled artifacts in `/tmp/torchinductor_{user}`. This is what appeared: 

**Run 1:**
* `torch.compile()`: doesn't produce any compiled artifact → lazy compilation
* **run 1, call 1:** creates `fxgraph`, `aotautograd`, and `inductor-spyre` with all the compiled elements, `inductor-spyre` contain all the spyre-specific artifacts
* **run1, call 2:** no additional artifacts → everything in memory and reused
* **run1, call 3 (uses a different shape)**: everything (all three `fxgraph`, `aotautograd` and `inductor-spyre`) recompiled with new artifacts→ probably due to `dynamic=False` (dynamic=True produced some error)

**Run 2 (expected some saved compilation reuse):** 
* **run2, call 1:** `inductor-spyre` content fully recompiled with new content (no reused of run1, call 1 `inductor-spyre` content at all), `fxgraph`, `aotautograd` reused
* **run2, call 2:** no additional artifacts
* **run2, call3:** also recompile the `inductor-spyre` instead of using run1, call3, `fxgraph`, `aotautograd` reused

**Next step:**
* Try multiple run of vllm on spyre and see if the speed-up is coherent with experiment above
* Timing the compilation time of each of `fxgraph`, `aotautograd` and `inductor-spyre`
* Investigate in `torch-spyre` if we need to set an environment variable to enable caching, or how to reuse cache

<details>

<summary> <b> test_matmul_caching.py</b> </summary>

```
import time
import torch
import torch_spyre  # noqa: F401 — registers the spyre device

import os
os.environ["TORCHINDUCTOR_FX_GRAPH_CACHE"] = "1"
os.environ["TORCHINDUCTOR_AUTOGRAD_CACHE"] = "1"

device = torch.device("spyre")

# --- Inputs ---

x  = torch.randn(1024, 1024, dtype=torch.float16, device=device)
y  = torch.randn(1024, 2048, dtype=torch.float16, device=device)

x2 = torch.randn(1024, 1024, dtype=torch.float16, device=device)
y2 = torch.randn(1024, 2048, dtype=torch.float16, device=device)

x3 = torch.randn(2048, 1024, dtype=torch.float16, device=device)
y3 = torch.randn(1024, 2048, dtype=torch.float16, device=device)

# --- Compile (no compilation happens here) ---

compiled_mm = torch.compile(torch.mm)
print("torch.compile() returned — no compilation yet")

# --- First call: full compilation pipeline ---

print("\n[1] First call — full compilation (shapes [1024,1024] x [1024,2048])")
t0 = time.perf_counter()
output1 = compiled_mm(x, y)
t1 = time.perf_counter()
print(f"    output shape : {output1.shape}")
print(f"    time         : {t1 - t0:.3f}s  (includes compilation)")

# --- Second call: same shapes, cache hit ---

print("\n[2] Second call — cache hit, same shapes")
t0 = time.perf_counter()
output2 = compiled_mm(x2, y2)
t1 = time.perf_counter()
print(f"    output shape : {output2.shape}")
print(f"    time         : {t1 - t0:.3f}s  (execution only)")

# --- Third call: different shapes, recompilation ---

print("\n[3] Third call — recompilation (shapes [2048,512] x [512,1024])")
t0 = time.perf_counter()
output3 = compiled_mm(x3, y3)
t1 = time.perf_counter()
print(f"    output shape : {output3.shape}")
print(f"    time         : {t1 - t0:.3f}s  (includes recompilation)")

print("\nDone.")

```

</details>

**Comment by sducouedic** [comment]
I am evaluating the effect of compile caching on real model run, on the micro model the warmup speedup (gained from `aotautograd` and `fxgraph` cache) is in the order of 10%. I am trying to hack the `inductor-spyre` caching to evaluate the speedup there if we had this part also that was also cached and reused.

**Comment by sducouedic** [comment]
Evaluating this PR to torch-spyre: https://github.com/torch-spyre/torch-spyre/pull/2397, first results on basic script with micro model show 80% warmup improvements thanks to cache reuse

**Comment by sducouedic** [comment]
Picking this back and trying to get a version that is up-to-date with main. An issue I am facing is invalidation of old artifacts as the code evolves, so somehow the key needs to integrate the version of the relevant piece of code. But currently torch-spyre doesn't have versioning, and dxp_standalone doesn't have a `--version`
